source/extensions/filters/http/json_to_metadata/filter.cc

```diff
@@ -512,6 +531,22 @@ Http::FilterTrailersStatus Filter::encodeTrailers(Http::ResponseTrailerMap&) {
   return Http::FilterTrailersStatus::Continue;
 }
 
+const FilterConfig* Filter::getConfig() const {
+  // Cached config pointer.
+  if (effective_config_) {
+    return effective_config_;
```

Comment:
Nice, this is a good use of  logical const method e.g. changing this internal cache